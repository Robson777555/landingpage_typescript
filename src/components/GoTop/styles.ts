import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;