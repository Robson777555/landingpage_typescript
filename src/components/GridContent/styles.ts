// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;