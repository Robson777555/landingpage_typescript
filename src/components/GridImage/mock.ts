interface GridItem {
  altText: string;
  srcImg: string;
}

interface MockData {
  title: string;
  description: string;
  grid: GridItem[];
  background?: boolean;
}

const mock: MockData = {
  background: false,
  title: '<b style="font-size: 2rem;">Tec</b>',
  description: 'Conhe√ßa as principais tecnologias usadas neste projeto:',
  grid: [
    {
      altText: 'JavaScript Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/6/6a/JavaScript-logo.png',
    },
    {
      altText: 'React Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg',
    },
    {
      altText: 'Next.js Logo',
      srcImg: 'https://assets.vercel.com/image/upload/v1662090959/front/nextjs/twitter-card.png',
    },
    {
      altText: 'Node.js Logo',
      srcImg: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Node.js_logo.svg',
    },
    {
      altText: 'Git Logo',
      srcImg: 'https://git-scm.com/images/logos/downloads/Git-Icon-1788C.png',
    },
  ],
};

export default mock;