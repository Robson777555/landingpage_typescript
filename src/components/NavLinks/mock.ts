interface Link {
  children: string;
  link: string;
  newTab?: boolean;
}

const mock: Link[] = [
  {
    children: 'ServiÃ§os',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'Tecnologias usadas',
    link: '#target2',
    newTab: false,
  },
  {
    children: 'Tecnologias essenciais',
    link: '#target3',
    newTab: false,
  },
];

export default mock;