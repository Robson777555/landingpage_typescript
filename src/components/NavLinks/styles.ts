import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.small};
    width: 100%;

    @media (min-width: 769px) {
      flex-direction: row;
      align-items: center;
      gap: ${theme.spacings.medium};
      width: auto;
    }

    /* Estilos para os links filhos */
    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      font-size: ${theme.fonts.sizes.medium};
      padding: ${theme.spacings.small} 0;
      transition: all 0.3s ease;
      font-family: ${theme.fonts.family.default};
      position: relative;
      outline: none;

      &:hover,
      &:focus {
        color: ${theme.colors.secondaryColor};
      }

      /* Mobile styles */
      @media ${theme.media.lteMedium} {
        border-bottom: 1px solid ${theme.colors.mediumGray};
        text-align: center;
        padding: ${theme.spacings.medium};
        width: 100%;
      }

      /* Acessibilidade - foco vis√≠vel */
      &:focus-visible {
        outline: 2px solid ${theme.colors.secondaryColor};
        outline-offset: 2px;
      }
    }
  `}
`;