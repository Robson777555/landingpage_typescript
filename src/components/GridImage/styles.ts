// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
