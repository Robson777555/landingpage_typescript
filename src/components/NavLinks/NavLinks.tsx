'use client';
import React from 'react';
import { MenuLink } from '../MenuLink/MenuLink';
import * as Styled from './styles';

interface LinkProps {
  children: string;
  link: string;
  newTab?: boolean;
}

interface NavLinksProps {
  links?: LinkProps[];
  onLinkClick?: () => void;
}

export const NavLinks = ({ links = [], onLinkClick }: NavLinksProps) => {
  return (
    <Styled.Container aria-label="Main menu" role="navigation">
      {links.map((link) => (
        <MenuLink
          key={link.link}
          {...link}
          onClick={onLinkClick}
          data-testid={`navlink-${link.link.replace('#', '')}`}
        />
      ))}
    </Styled.Container>
  );
};