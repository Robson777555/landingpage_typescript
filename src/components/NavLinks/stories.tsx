import { Meta, StoryObj } from "@storybook/react";
import { NavLinks } from "./NavLinks";
import mock from "./mock";

const meta: Meta<typeof NavLinks> = {
  title: "Components/NavLinks",
  component: NavLinks,
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    links: {
      control: {
        type: "object",
      },
    },
    onLinkClick: {
      action: "clicked",
    },
  },
  args: {
    links: mock,
  },
};

export default meta;

type Story = StoryObj<typeof NavLinks>;

export const Default: Story = {};

export const MobileView: Story = {
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  },
};

export const Empty: Story = {
  args: {
    links: [],
  },
};
