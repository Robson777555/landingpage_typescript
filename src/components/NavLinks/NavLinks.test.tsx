import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { NavLinks } from './NavLinks';
import { theme } from '../../styles/theme';
import { describe, it, expect, vi } from 'vitest';
import mock from './mock';

describe('<NavLinks />', () => {
  const renderNavLinks = (props = {}) => {
    const defaultProps = {
      links: mock,
    };

    return render(
      <ThemeProvider theme={theme}>
        <NavLinks {...defaultProps} {...props} />
      </ThemeProvider>
    );
  };

  it('should render correct number of links with proper attributes', () => {
    renderNavLinks();

    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(mock.length);

    mock.forEach((link, index) => {
      expect(links[index]).toHaveAttribute('href', link.link);
      expect(links[index]).toHaveTextContent(link.children);
    });
  });

  it('should call onLinkClick when a link is clicked', () => {
    const mockClick = vi.fn();
    renderNavLinks({ onLinkClick: mockClick });

    fireEvent.click(screen.getByText(mock[0].children));
    expect(mockClick).toHaveBeenCalled();
  });

  it('should not render links when empty array is provided', () => {
    renderNavLinks({ links: [] });
    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });

  it('should match snapshot with links', () => {
    const { container } = renderNavLinks();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot without links', () => {
    const { container } = renderNavLinks({ links: [] });
    expect(container.firstChild).toMatchSnapshot();
  });
  it(' Snapshpt page ', () => {
    const { container } = renderNavLinks();
    expect(container.firstChild).toMatchSnapshot();
  });
});