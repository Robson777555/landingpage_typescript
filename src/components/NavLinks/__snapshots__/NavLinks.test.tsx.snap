// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavLinks /> >  Snapshpt page  1`] = `
<nav
  aria-label="Main menu"
  class="sc-hKDTPf kOFeWo"
  role="navigation"
>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target1"
    target="_self"
  >
    Serviços
  </a>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target2"
    target="_self"
  >
    Tecnologias usadas
  </a>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target3"
    target="_self"
  >
    Tecnologias essenciais
  </a>
</nav>
`;

exports[`<NavLinks /> > should match snapshot with links 1`] = `
<nav
  aria-label="Main menu"
  class="sc-hKDTPf kOFeWo"
  role="navigation"
>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target1"
    target="_self"
  >
    Serviços
  </a>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target2"
    target="_self"
  >
    Tecnologias usadas
  </a>
  <a
    class="sc-bXdtCk kDKUyx"
    href="#target3"
    target="_self"
  >
    Tecnologias essenciais
  </a>
</nav>
`;

exports[`<NavLinks /> > should match snapshot without links 1`] = `
<nav
  aria-label="Main menu"
  class="sc-hKDTPf kOFeWo"
  role="navigation"
/>
`;
